### PR TITLE
QTY-1376: bring accounts_report up to date to work  

### PR DESCRIPTION
**Pull request by adiaz04**
[QTY-1376](https://strongmind.atlassian.net/browse/QTY-1376)

## Purpose 
Upgrading to ruby 2.6.10 created an issue in the accounts_report gem which was using instance variables from the actual csv gem, updated the code to work with 2.6.10

## Approach 
Looked up the incremental fixed that instructure did for this and cherry-picked those commits.

## Testing
locally and on new-id-sandbox, i generated all the possible reports.
)

## Purpose 
Upgrading to ruby 2.6.10 created an issue in the accounts_report gem which was using instance variables from the actual csv gem, updated the code to work with 2.6.10

## Approach 
Looked up the incremental fixed that instructure did for this and cherry-picked those commits.

## Testing
locally and on new-id-sandbox, i generated all the possible reports.

### CODE REVIEW

File: gems/plugins/account_reports/lib/account_reports/report_helper.rb [hunk 1]
```diff
@@ -239,13 +244,163 @@ def generate_and_run_report(headers = nil, extention = 'csv')
...
+    updates[:current_line] = current_line if account_report.current_line < current_line
+    updates[:progress] = progress if account_report.progress < progress
+    unless updates.empty?
+      Shackles.activate(:master) do
+        AccountReport.where(id: account_report).where("progress <?", progress).update_all(updates)
+      end
+    end
+  end
+
+  def last_account_report_runner?(account_report)
```
[the remark this answers is not in the document]
Comment (adiaz04) [reply gems/plugins/account_reports/lib/account_reports/report_helper.rb:382]: cherry picked from instructure.

File: lib/api/v1/outcome_results.rb [hunk 2]
```diff
@@ -191,8 +191,12 @@ def serialize_rollup_score(score)
     }
   end
 
-  def outcome_results_rollups_csv(rollups, outcomes, outcome_paths)
-    CSV.generate do |csv|
+  def outcome_results_rollups_csv(current_user, context, rollups, outcomes, outcome_paths)
```
[the remark this answers is not in the document]
Comment (adiaz04) [reply lib/api/v1/outcome_results.rb:194]: cherry picked from instructure.

[the remark this answers is not in the document]
Comment (adiaz04) [reply lib/api/v1/outcome_results.rb:194]: cherry picked from instructure.

File: gems/plugins/account_reports/lib/account_reports/report_helper.rb [hunk 3]
```diff
@@ -239,13 +244,163 @@ def generate_and_run_report(headers = nil, extention = 'csv')
...
+  def runner_aborted?(report_runner)
+    if report_runner.reload.workflow_state == 'aborted'
+      report_runner.delete_account_report_rows
+      true
+    else
+      false
+    end
+  end
+
+  def update_parallel_progress(account_report: @account_report, report_runner:)
```
[the remark this answers is not in the document]
Comment (adiaz04) [reply gems/plugins/account_reports/lib/account_reports/report_helper.rb:364]: cherry picked from instructure.

**Review by agtorre3 — APPROVED** [review]
Since this cherry picks a lot of stuff from instructure's repo, I would maybe deploy to one instance first to test export for real grade data

**Review by CJAdeszko — APPROVED** [review]

**Pull request merged**